### DISCUSSION TITLE
Anybody getting lost in lec.14 Creating your own libraries?

### DISCUSSION THREAD

**Discussion by yashsharma22003** [Cyfrin/foundry-full-course-cu#2179] [General]
I was watching Solidity smart contract development series and was just confused and lost on this lecture 14 of the series anyone else facing the same problem?
Would appreciate if someone could explain me in layman terms

**Comment by EngrPips** [comment]
Please share the link to what you are watching. @yashsharma22003

**Comment by SwarnenduG07** [comment]
where you are stucked? And in which part?

**Comment by yashsharma22003** [comment]
https://updraft.cyfrin.io/courses/solidity/
Section 3
Video no .14

**Comment by yashsharma22003** [comment]
https://updraft.cyfrin.io/courses/solidity
Sorry this is the working link to the course I am referring to

**Reply by EngrPips** [reply]
You mean this lesson https://updraft.cyfrin.io/courses/solidity/fund-me/create-solidity-library, correct?

**Reply by yashsharma22003** [reply]
Yeah

**Reply by EngrPips** [reply]
That's solidity library he was explaining. Library in solidity is just like contracts, but it is written with the `library` keyword instead of `contract`. They don't have any state variables, and you can send them ether. We usually always create a library when we have a set of commonly utilized functionality, and we don't want to always have to rewrite that functionality for every single contract that utilizes that common functionality. So we just put the functionality in a solidity library, and we can always import that library to make use of it. You can read more about Solidity library here https://www.geeksforgeeks.org/solidity-libraries/

**Reply by yashsharma22003** [reply]
Spot on!
Thanks for the clarification ☺️

**Reply by EngrPips** [reply]
You are welcome, Friend.
